java/ql/lib/semmle/code/java/frameworks/spring/SpringHttp.qll

```diff
@@ -140,3 +143,179 @@ private class SpringHttpFlowStep extends SummaryModelCsv {
       ]
   }
 }
+
+private predicate specifiesContentType(SpringRequestMappingMethod method) {
+  method.getProducesExpr().(ArrayInit).getSize() != 0 or
+  not method.getProducesExpr() instanceof ArrayInit
```

Comment:
This seems off. If `not exists(method.getProducesExpr())` then this predicate will hold?!

Reply:
Or does `exists(method.getProducesExpr())` always hold?

Reply:
Simplified this to `exists(method.getAProducesExpr())`